### PR TITLE
Multiple bug fixes

### PR DESCRIPTION
**Pull request by NeuralAlchemist**
Bug fixes
- FormFooter
  - commentForm will now send in postid as well which is used to label the input tags correctly to each post. Otherwise, no matter where the files are selected, it gets added to the latest post
  - length is checked against undefined instead of null
- #73 ; #59 ; #54
- Likes can now be added/removed to posts and comments without a reload

## User Story 68
Uploading the same video displays a blank home page (issue #59)
(Tested thoroughly with a lot of different videos (large,medium and small size)

## User Story 70
Updating a post with image, will remove the image when it is posted (after update) (issue #54)
(Testing performed on postman) 

## User Story 71
After commenting/posting a video, it is sometimes not found (issue #73)
Screenshot 2021-05-16 at 17 56 23

Saw a possible bug with these changes , this is #74

### CODE REVIEW

File: frontend/src/components/FormFooter.jsx [hunk 1]
```diff
@@ -36,9 +36,9 @@ export default function FormFooter({
         <button
           className="button-post"
           onClick={handleSubmit}
-          disabled={length === null || length === 0}
+          disabled={length === undefined || length === 0}
         >
-          {buttonText}
+          {buttonText.includes("comment") ? "comment" : "post"}
```
Comment (eeels22) [inline frontend/src/components/FormFooter.jsx]: Please could these be capitalised?

`{buttonText.includes("comment") ? "Comment" : "Post"}`

File: frontend/src/components/comments/CommentCard.jsx [hunk 2]
```diff
@@ -24,19 +29,20 @@ export default function CommentCard({
...
   };
 
-  async function updateComment(updatedComment) {
+  async function updateComment(contentText) {
     try {
-      await CommentsApi.updateComment(postId, comment.id, updatedComment);
+      let formData = new FormData();
+      console.log(contentText);
+      formData.append("text", contentText);
+      console.log(formData);
```
Comment (eeels22) [inline frontend/src/components/comments/CommentCard.jsx]: Console.logs could be removed unless you think it's handy to leave in for debugging related issues.

File: frontend/src/components/posts/PostCard.jsx [hunk 3]
```diff
@@ -15,21 +15,28 @@ export default function PostCard({ post, currentUser, onDeleteClick }) {
...
 
   // Methods
-  async function updatePost(updatedPost) {
+  async function updatePost(contentText) {
     try {
-      await PostsApi.updatePost(updatedPost, post.id);
+      let formData = new FormData();
+      console.log(contentText);
+      formData.append("text", contentText);
+      console.log(formData);
```
Comment (eeels22) [inline frontend/src/components/posts/PostCard.jsx]: Console logs could be removed unless you think it's better to keep in for now

File: frontend/src/components/posts/PostForm.jsx [hunk 4]
```diff
@@ -24,8 +24,9 @@ export default function PostForm({ onSubmit, onSubmitMedia }) {
       // Invoke the passed in event callback
       onSubmit({ contentText: contentText });
       // Clear the input field
-      setContentText("");
+
```
Comment (eeels22) [inline frontend/src/components/posts/PostForm.jsx]: Could delete these lines

File: frontend/src/pages/posts/PostsPage.jsx [hunk 5]
```diff
@@ -49,6 +50,7 @@ export default function PostsPage() {
   }
 
   useEffect(() => {
+    console.log("GETTING ALL POSTS");
```
Comment (eeels22) [inline frontend/src/pages/posts/PostsPage.jsx]: Console log - delete if no longer needed

**Review by eeels22 — CHANGES_REQUESTED** [review]
Neat fixes Yash, good idea with the timeout! Just left a request to capitalise the button text on FormFooter and flagged a few console.logs.

**Review by janko101 — APPROVED** [review]
It looks great...cheers for likes not reloading

**Review by eeels22 — APPROVED** [review]
Great work Yash!

**Pull request merged**